services/supervisor/src/transactions.rs

```diff
@@ -211,26 +225,20 @@ impl SupervisorInterface for Supervisor {
             .validator_id(author)
             .ok_or(Error::UnknownAuthor)?;
 
-        let entry = schema
-            .pending_proposal()
-            .get()
-            .ok_or_else(|| Error::ConfigProposeNotRegistered)?;
-
         // Verifies that this config proposal is registered.
-        if entry.propose_hash != vote.propose_hash {
-            return Err(Error::ConfigProposeNotRegistered.into());
-        }
-
-        let config_propose = entry.config_propose;
+        let proposal = schema
+            .pending_proposals()
+            .values()
+            .find(|proposal| proposal.propose_hash == vote.propose_hash)
+            .ok_or_else(|| Error::ConfigProposeNotRegistered)?;
         // Verifies that we didn't reach the deadline height.
-        if config_propose.actual_from <= blockchain_height {
+        if proposal.config_propose.actual_from <= blockchain_height {
```

Comment:
There is an interesting situation here. `Supervisor` is trying to accept config for the _next_ block in `before_commit`. However, votes for a _current_ block aren't accepted. So, at the height of expected acceptance, proposal enters a weird state when it's not yet removed, but already can't be voted for.